module/modmanager/manager.go

```diff
@@ -39,14 +39,17 @@ var (
 
 // NewManager returns a Manager.
 func NewManager(parentAddr string, logger golog.Logger, options modmanageroptions.Options) modmaninterface.ModuleManager {
-	return &Manager{
-		logger:                  logger,
-		modules:                 map[string]*module{},
-		parentAddr:              parentAddr,
-		rMap:                    map[resource.Name]*module{},
-		untrustedEnv:            options.UntrustedEnv,
-		removeOrphanedResources: options.RemoveOrphanedResources,
+	mgr := &Manager{
+		logger:       logger,
+		modules:      map[string]*module{},
+		parentAddr:   parentAddr,
+		rMap:         map[resource.Name]*module{},
+		untrustedEnv: options.UntrustedEnv,
 	}
+	if options.RemoveOrphanedResources != nil {
+		mgr.removeOrphanedResources = options.RemoveOrphanedResources
+	}
```

Comment:
Assigning a nil would't cause a direct problem, so this check doesn't change anything. But rather I was suggesting this should error if the function wasn't provided in the options.

The later checks mean it at least won't panic though, and I suppose there could be test cases where you don't want to have a remove-orphan function, so that's probably fine. I will leave it up to you. The check here is still doing nothing though, so please either revert this specific change (just directly assign the option as before) or make this return an error if it's nil.

Reply:
You're right this doesn't really guard anything; I'll just set `removeOrphanedResources` directly (revert the change).
 
I like guarding throughout the manager and not returning an error in creation. Multiple tests create a module manager with no concept of a resource manager, and therefore no need for a `RemoveOrphanedResources` function. Rather than pass a "dummy" one in that asserts the function is not called, I'd rather allow the creation and use of a module manager without the function when we don't need to assert anything about a related resource manager.